custom-transformer/src/main/scala/com/flipkart/fdp/ml/IfZeroVector.scala

```diff
@@ -0,0 +1,78 @@
+package com.flipkart.fdp.ml
+
+import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.param.{Params, Param, ParamMap}
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.mllib.linalg.{DenseVector, SparseVector, Vector, VectorUDT, Vectors}
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.StructType
+
+
+trait IfZeroVectorParams extends Params {
+  final val inputCol: Param[String] = new Param[String](this, "inputCol", "input column name")
+  final def getInputCol: String = $(inputCol)
+
+  final val outputCol: Param[String] = new Param[String](this, "outputCol", "output column name")
+  setDefault(outputCol, uid + "__output")
+  final def getOutputCol: String = $(outputCol)
+
+  final val thenSetValue: Param[String] = new Param[String](this, "string constant", "a string constant")
```

Comment:
Not doing this at the moment. We'll take this up when we plan to generalize this.